packages/SwingSet/docs/virtual-objects.md

```diff
@@ -31,50 +31,56 @@ A `kindHandle` is a type of durable object that can be used to identify the kind
+In either case, the individual behavior functions must have the signature:
 
-The `finish` parameter is optional. It is a function that, if present, will be called exactly once as part of instance initialization.  It will be invoked _immediately_ after the `actualize` function for that instance is called for the very first time.  In other words, it will be called after the instance per se exists but before that instance is returned from the maker function to whoever requested its creation.  `finish` is passed two parameters: the VDO's state (exactly as passed to the `actualize` function) and the VDO itself. The `finish` function can modify the object's state in the context of knowing the object's identity, and thus can be used in cases where a validly initialized instance requires it to participate in some kind of cyclical object graph with other VDOs.  It can also be used, for example, to register the object with outside tracking data structures, or do whatever other post-creation setup is needed for the object to do its job.  In particular, if one or more of the object's methods need to refer to the object itself (for example, so it can pass itself to other objects), the `finish` function provides a way to capture that identity as part of the object's state.
+  `methodname(context, ...args) { ...`
+
+where `context` describes the invocation context of the method and `...args` are whatever arguments were passed to the method when it was invoked.  In the case of a single facet VDO, `context` will take the form `{ state, self }`, where `state` is the VDO state and `self` is a reference to the VDO itself.  In the case of a multi-facet VDO, `context` will instead be `{ state, facets }`, where `facets` is an object whose named properties are the facets of the VDO.
+
+The `options` parameter is optional. It provides additional parameters to characterize the VDO.  Currently there is only one supported option, `finish`, though we anticipate more options may be added in future versions of the API.
+
+The `finish` option is a function that, if present, will be called at the end of instance initialization.  It will be invoked after the VDO is created but before it is returned from the maker function.  `finish` is passed two parameters: (1) the VDO's state and (2) the VDO itself or an object containing its facets.  This is essentially the same two elements that are in the `context` parameter that is passed to method behaviors.  The `finish` function can modify the object's state in the context of knowing the object's (or its facets) identity, and thus can be used in cases where a validly initialized instance requires it to participate in some kind of cyclical object graph with other VDOs.  It can also be used, for example, to register the object with outside tracking data structures, or do whatever other post-creation setup is needed for the object to do its job.
```

Comment:
Should we maybe give it the same `context` object as the `behavior` methods get? I don't think we explored this when talking about `finish`, but now I'm wondering if it would be more uniform. Not a big deal either way, for sure.

Reply:
The same thought occurred to me as I was writing it, but ultimately I decided that the gratuitous destructuring didn't really pay for itself.  However, that's not a strongly held belief on my part and I'm happy to change it to work the other way if that's the consensus.

Reply:
My inclination is the other way, to use the same context object. Not because the destructuring is compelling, but so there's one less thing to learn.

Reply:
OK, done.